src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -4075,6 +4075,12 @@ public virtual void Dispose()
                 }
                 Validate();
             }
+
+#if FEATURE_PERFTRACING
+            // Remove the listener from the EventPipe dispatcher. EventCommand.Update with enable==false removes it.
+            // Have to send outside of EventListenersLock, or we risk deadlocks.
```

Comment:
Is this comment true any longer? I'd expect you can call SendCommand() with the EventListenerLock held and it produces no deadlock.